### PR TITLE
106 fractionalcurrent

### PR DESCRIPTION
**Pull request by dyldonahue**
## Issue ticket number and issue description
#106 fractional current

I multiplied pack current by 10 to allow for single decimal precision. I then divided it by 10 for all calcs (that did not care about decimal) to ensure all existing calculations were unaffected. 

Please double check that my manipulation of the value is done correctly and that I didn't miss any instances. where it is used

I also did not: 

divide packCurrent by 10 when broadcasting (idk if we wanna see decimal for this or not)
adjust open cell voltage calc to use this
add this value to charging stuff (not merged in yet)

lmk if any of those should be done here

### CODE REVIEW

**Review by mccauleyma — CHANGES_REQUESTED** [review]
Additional to my comments, you missed the current comparison in Analyzer::calcOpenCellVoltage()

File: shepherd_bms/src/main.cpp [hunk 1]
```diff
@@ -138,7 +138,7 @@ void broadcastCurrentLimit(AccumulatorData_t *bmsdata)
 		BoostState = BOOST_STANDBY;
 	}
 	//Transition to boosting
-	if(bmsdata->packCurrent > (int16_t)bmsdata->contDCL && BoostState == BOOST_STANDBY)
+	if((bmsdata->packCurrent)/10 > (int16_t)bmsdata->contDCL && BoostState == BOOST_STANDBY)
```
[the remark this answers is not in the document]
Comment (dyldonahue) [reply shepherd_bms/src/main.cpp:141]: contDCL was already casted that way, even though it was already a int16. Will remove if that was never meant to be the case

[the remark this answers is not in the document]
Comment (dyldonahue) [reply shepherd_bms/src/main.cpp:141]: The divide vs multiply by 10 was the debate, but yeah I agree so I will flip for all instances

**Review by mccauleyma — CHANGES_REQUESTED** [review]
Still missing the open cell voltage calc

**Review by mccauleyma — APPROVED** [review]
:)

**Review by nwdepatie — APPROVED** [review]
Seal of approval(but the other seal)

**Pull request merged**